### PR TITLE
geozero-shp release with deprecation message

### PR DESCRIPTION
**Pull request by pka**
Should be released & merged before #233

### CODE REVIEW

**Review by kylebarron — APPROVED** [review]

**Review by michaelkirk — CHANGES_REQUESTED** [review]
Can you include a module level declaration so that people can see it in their compiler output, since people often don't read documentation:

Something like:
```
#![deprecated(note = "The geozero_shp crate has been deprecated. Instead, use geozero's built-in shapefile feature")]
```

File: geozero-shp/Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "geozero-shp"
-version = "0.5.0"
```
Comment (michaelkirk) [inline geozero-shp/Cargo.toml:3]: I think this should be a patch bump - we're not breaking anything right?

**Comment by pka** [comment]
Test fails because of clippy complaints... Will force merge it when both PRs are ready.

**Review by michaelkirk — APPROVED** [review]

**Comment by pka** [comment]
Note to self: Release to crates.io just after next geozero release!

**Comment by michaelkirk** [comment]
Will we have to pin to a specific (old) version of geozero for this geozero_shp release? Currently it's using "workspace"

**Pull request merged**